service/src/main/java/com/TWCC/api/ResponseParser.java

```diff
@@ -375,6 +380,81 @@ private int extractAgeLimit(Map<String, Object> eventMap) {
         return ageLimit;
     }
 
+    /**
+     *
+     * @param jsonString
+     * @return
+     */
+    private String extractCategory(Map<String, Object> eventMap) {
+
+        String category = "";
+
+        if (eventMap.get("classifications") == null) {
+            return category;
+        }
+
+        @SuppressWarnings("unchecked")
+        ArrayList<Object> classificationList = (ArrayList<Object>) eventMap.get(
+            "classifications"
+        );
+
+        if (classificationList.size() == 0) {
+            return category;
+        }
+
+        @SuppressWarnings("unchecked")
+        Map<String, Object> categoryMap = (HashMap<String, Object>) classificationList.get(
+            0
+        );
+
+        if (categoryMap.get("segment") == null) {
+            return category;
+        }
+
+        @SuppressWarnings("unchecked")
+        String segment = (
+                            (TextNode)
+                                (
+                                    (HashMap<String, Object>)
+                                        categoryMap.get("segment")
+                                ).get("name")
+                        ).toString();
+
+        category += segment;
+
+        if (categoryMap.get("genre") == null) {
+            return category;
+        }
+
+        @SuppressWarnings("unchecked")
+        String genre = (
+                            (TextNode)
+                                (
+                                    (HashMap<String, Object>)
+                                        categoryMap.get("genre")
+                                ).get("name")
+                        ).toString();
+
+        category += "," + genre;
+
+        if (categoryMap.get("subGenre") == null) {
+            return category;
+        }
+
+        @SuppressWarnings("unchecked")
+        String subGenre = (
+                            (TextNode)
+                                (
+                                    (HashMap<String, Object>)
+                                        categoryMap.get("subGenre")
+                                ).get("name")
+                        ).toString();
+
+        category += "," + subGenre;
```

Comment:
Is there only one `genre` and `subGenre` that we're adding to category? If I remember correctly, you showed me multiple genres and subgenres, but the code here only grabs one value?

Reply:
The response's combines all the the category-related info into an attribute named `classifications`, and all the sample data I have seen only has one entry in the `classification` attribute, which contains one `segment`, one `genre`, and one `subGenre`. `classification` entries also contains `type` and `subType` attributes, but I deemed these attributes irrelevant to the categories we are looking for. 

Here is a sample `classification` attribute:
```
"classifications": [
                    {
                        "primary": true,
                        "segment": {
                            "id": "KZFzniwnSyZfZ7v7nE",
                            "name": "Sports"
                        },
                        "genre": {
                            "id": "KnvZfZ7vAde",
                            "name": "Basketball"
                        },
                        "subGenre": {
                            "id": "KZazBEonSMnZfZ7vFJA",
                            "name": "NBA"
                        },
                        "type": {
                            "id": "KZAyXgnZfZ7v7l1",
                            "name": "Group"
                        },
                        "subType": {
                            "id": "KZFzBErXgnZfZ7vA7d",
                            "name": "Team"
                        },
                        "family": false
                    }
                ]
```

Reply:
I see. Weren't we going with type and subtype too?

Reply:
Resolved after discussing not as useful values to be included